### PR TITLE
Catch KeyNotFoundException

### LINKED ISSUE #2012: [BUG] Unhandled Exception: KeyNotFoundException
### Prerequisites

- [X] I made sure I am running the latest [development build](https://ci.appveyor.com/project/ORelio/minecraft-console-client/build/artifacts)
- [X] I tried to [look for similar issues](https://github.com/MCCTeam/Minecraft-Console-Client/issues?q=is%3Aissue) before opening a new one
- [X] I have set `debugmessages=true` in config to diagnose my issue
- [X] I have redacted session tokens and passwords before attaching screenshots

### Minecraft Version

1.18.2

### Console Client Version

Not latest sadly

### Expected Behavior

Catch the exception and not have fatal crash.

### Actual Behavior

MCC has crashed for some reason after trying to relog. (with its auto relog module)

### Steps to Reproduce the bug

I'm not sure how to reproduce this bug sadly. 

### Attach screenshot here (If applicable)

_No response_

### Anythings that could help diagnosing the bug

```txt
Restarting Minecraft Console Client...
Cached session is invalid or expired.

Unhandled Exception:
System.Collections.Generic.KeyNotFoundException: The given key 'access_token' was not present in the dictionary.
  at System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) [0x00021] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at MinecraftClient.Protocol.Microsoft.RequestToken (System.String postData) [0x00058] in <b2e887d387e84817ab9b4c527f3b4c9c>:0 
  at MinecraftClient.Protocol.Microsoft.RefreshAccessToken (System.String refreshToken) [0x00010] in <b2e887d387e84817ab9b4c527f3b4c9c>:0 
  at MinecraftClient.Protocol.ProtocolHandler.MicrosoftLoginRefresh (System.String refreshToken, MinecraftClient.Protocol.Session.SessionToken& session) [0x00000] in <b2e887d387e84817ab9b4c527f3b4c9c>:0 
  at MinecraftClient.Program.InitializeClient () [0x000aa] in <b2e887d387e84817ab9b4c527f3b4c9c>:0 
  at MinecraftClient.Program+<>c__DisplayClass11_0.<Restart>b__0 () [0x0007a] in <b2e887d387e84817ab9b4c527f3b4c9c>:0 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00017] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x0008d] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x00031] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Threading.ThreadHelper.ThreadStart () [0x0000b] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.Collections.Generic.KeyNotFoundException: The given key 'access_token' was not present in the dictionary.
  at System.Collections.Generic.Dictionary`2[TKey,TValue].get_Item (TKey key) [0x00021] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at MinecraftClient.Protocol.Microsoft.RequestToken (System.String postData) [0x00058] in <b2e887d387e84817ab9b4c527f3b4c9c>:0 
  at MinecraftClient.Protocol.Microsoft.RefreshAccessToken (System.String refreshToken) [0x00010] in <b2e887d387e84817ab9b4c527f3b4c9c>:0 
  at MinecraftClient.Protocol.ProtocolHandler.MicrosoftLoginRefresh (System.String refreshToken, MinecraftClient.Protocol.Session.SessionToken& session) [0x00000] in <b2e887d387e84817ab9b4c527f3b4c9c>:0 
  at MinecraftClient.Program.InitializeClient () [0x000aa] in <b2e887d387e84817ab9b4c527f3b4c9c>:0 
  at MinecraftClient.Program+<>c__DisplayClass11_0.<Restart>b__0 () [0x0007a] in <b2e887d387e84817ab9b4c527f3b4c9c>:0 
  at System.Threading.ThreadHelper.ThreadStart_Context (System.Object state) [0x00017] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x0008d] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state) [0x00031] in <12b418a7818c4ca0893feeaaf67f1e7f>:0 
  at System.Threading.ThreadHelper.ThreadStart () [0x0000b] in <12b418a7818c4ca0893feeaaf67f1e7f>:0
```

### Device

Laptop

### Operating System

Linux

### Server Address (If applicable)

_No response_

### PR DESCRIPTION
**Pull request by udu3324**
This commit fixes #2012 I think.

### CODE REVIEW

**Comment by ReinforceZwei** [comment]
Hi, thank you for your contribution first. Your PR should fixes the `KeyNotFound` exception. However, it would be better to catch the error in [`MicrosoftBrowserLogin`](https://github.com/MCCTeam/Minecraft-Console-Client/blob/b3cc2351ee14d8e4fadc6f0e6115478872143c48/MinecraftClient/Protocol/ProtocolHandler.cs#L498) and [`MicrosoftLoginRefresh`](https://github.com/MCCTeam/Minecraft-Console-Client/blob/b3cc2351ee14d8e4fadc6f0e6115478872143c48/MinecraftClient/Protocol/ProtocolHandler.cs#L514) method, like how [`MojangLogin`](https://github.com/MCCTeam/Minecraft-Console-Client/blob/b3cc2351ee14d8e4fadc6f0e6115478872143c48/MinecraftClient/Protocol/ProtocolHandler.cs#L380) and [`MicrosoftMCCLogin`](https://github.com/MCCTeam/Minecraft-Console-Client/blob/b3cc2351ee14d8e4fadc6f0e6115478872143c48/MinecraftClient/Protocol/ProtocolHandler.cs#L467) method does. Then you would be able to return an unsuccessful login result back to the login process and let other parts (e.g. `AutoRelog`) to handle the failure.

**Comment by udu3324** [comment]
> Hi, thank you for your contribution first. Your PR should fixes the `KeyNotFound` exception. However, it would be better to catch the error in [`MicrosoftBrowserLogin`](https://github.com/MCCTeam/Minecraft-Console-Client/blob/b3cc2351ee14d8e4fadc6f0e6115478872143c48/MinecraftClient/Protocol/ProtocolHandler.cs#L498) and [`MicrosoftLoginRefresh`](https://github.com/MCCTeam/Minecraft-Console-Client/blob/b3cc2351ee14d8e4fadc6f0e6115478872143c48/MinecraftClient/Protocol/ProtocolHandler.cs#L514) method, like how [`MojangLogin`](https://github.com/MCCTeam/Minecraft-Console-Client/blob/b3cc2351ee14d8e4fadc6f0e6115478872143c48/MinecraftClient/Protocol/ProtocolHandler.cs#L380) and [`MicrosoftMCCLogin`](https://github.com/MCCTeam/Minecraft-Console-Client/blob/b3cc2351ee14d8e4fadc6f0e6115478872143c48/MinecraftClient/Protocol/ProtocolHandler.cs#L467) method does. Then you would be able to return an unsuccessful login result back to the login process and let other parts (e.g. `AutoRelog`) to handle the failure.

Sadly, I'm not sure how to do that. LoginResponse RequestToken does not return anything and the things you mentioned do. 
But based on [other commits](https://github.com/MCCTeam/Minecraft-Console-Client/commit/ed452cf632fcb855e60d99d58aa69ef04c0b1b88), catching it seems to work and AutoRelog does not need to handle it.

**Comment by ORelio** [comment]
Hi @ReinforceZwei and @udu3324,
Is this pull request still applicable or was the bug fixed elsewhere?

**Comment by ReinforceZwei** [comment]
I'm less motivated on this project because I don't play Minecraft now. But I think the patch here is not applicable (or acceptable) as I mentioned above. Good news is I don't see more complaint about this issue, so I guess key missing was some temporary problem on MS side. Maybe I'll close this PR for now.

File: MinecraftClient/Protocol/MicrosoftAuthentication.cs [hunk 1]
```diff
@@ -69,21 +69,33 @@ private static LoginResponse RequestToken(string postData)
...
+                    string email = jsonPayload.Properties["email"].StringValue;
+                    return new LoginResponse()
+                    {
+                        Email = email,
+                        AccessToken = accessToken,
+                        RefreshToken = refreshToken,
+                        ExpiresIn = expiresIn
+                    }; 
+                }
+                catch {Exception e}
```
Comment (JesperKuipers) [inline MinecraftClient/Protocol/MicrosoftAuthentication.cs:90]: should be:
`catch (Exception e)`

**Pull request closed without merging**